Fix incorrect weather icons and states: 'Clear Night' shown during day and 'Sunny' shown at night

Ay1tsMe opened the pull request:
This PR aims to solve the following problems that have been addressed in these issues:
https://github.com/bremor/bureau_of_meteorology/issues/177
https://github.com/bremor/bureau_of_meteorology/issues/163
https://github.com/bremor/bureau_of_meteorology/issues/74

Because the BOM API is inconsistent with its reportings sometimes, there are cases where the wrong weather icon/state can show at different parts of the day. These are the following cases that this PR aims to fix:
1. Forecast shows 'Sunny' at nighttime: 
There are times where the API will report the `icon_descriptor` as 'sunny' during the night. This has been fixed by checking the `is_night` boolean and overwriting the `icon_descriptor` to 'clear' if it is showing 'sunny' or 'mostly_sunny'.

2. Forecast shows 'Clear Night' during the day:
Sometimes the API will report the `icon_descriptor` during the day as 'clear'. Because home assistants weather integration uses 'clear' to represent 'clear-night', the integration will sometimes incorrectly report it being nighttime.  This has been fixed by checking the `is_night` boolean and overwriting the `icon_descriptor` to 'sunny' if it is showing 'clear'.

Ay1tsMe committed:
documentation for potential fixes

Ay1tsMe committed:
Added problem

Ay1tsMe committed:
Add logic to change 'sunny/mostly-sunny' to 'clear' if nighttime

Ay1tsMe committed:
Remove logging

Ay1tsMe committed:
deleted todo list

Ay1tsMe committed:
change icon_descriptor to 'sunny' if 'clear' and nighttime for hourly

systemtester commented:
Have tested hourly and daily forecast objects with multiple locations across the is_night changeover period.  Incidentally it looks like BOM sets is_night=true at 6pm for the timezone the location is in.  I assume they set it back to false at 6am but am not sure.  I'd have to capture the api for a 12h period to check that and they may have seasonal logic that drives the true/false timing.  If they do its a bit clunky.

Those locations being:

1. Tamworth / r67r0r (sunny) +10GMT
2. North Coogee / qd636j (mostly_sunny)  +8GMT

Spotted another small logic error for locations not in the Home Assistant timezone and how the hourly forecast presents the time.  ie.  I am in AEST (+10) and North Coogee is WST (+8).  While the forecast now shows the correct icon/description fixed in this PR the hourly times are presented in my local instance (+10) when I know that it will not be sunny at 7pm in North Coogee as shown in the screengrab below.  If you can make those adjustments @Ay1tsMe and resubmit please tag Simon to have these merged.  Please also update the readme.md with 1.3.6 comments to help him out.

![image](https://github.com/user-attachments/assets/d41986f1-9a5c-4ab7-ae88-d1fd6dfb7dfc)

Ay1tsMe committed:
Remove timezone offset

Ay1tsMe committed:
Merge pull request #1 from Ay1tsMe/timezone

Remove timezone offset

Ay1tsMe committed:
Documentation for release

Ay1tsMe commented:
@systemtester @Makin-Things Should be good now. Just need to test the timezone change I implemented. It should now show the local time of whatever weather station you are viewing instead of it trying to convert to HA's local time. I added multiple weather stations across Australia and they all seem to show the correct time along with the correct weather icon/state.

Makin-Things commented:
HI @Ay1tsMe thanks fo the PR. I was going to take a detailed look at it yesterday, but @systemtester  got in first. I have to agree that the bom have some very simplistic is night logic that doesn't work well. I am about to have a quick look before merging and creating a pre-release for further testing by others.

systemtester commented:
Have tested the timezone issue I noted. and it works well now across AEST, CST, WST.  Might be one to keep an eye out for when Daylight savings kicks in though.  Thanks for making that adjustment @Ay1tsMe.  Take a look at my comment above about d.get.  Not a biggie that its not done but worth noting.  I have already tested the adjustments to fix incorrect data.

All good by me but understand if you want to keep it pre-release a little longer @Makin-Things

Makin-Things commented:
I'll wait another day or so. SO probably change it either tomorrow night or Thurs. @systemtester thanks for the reminder though as I am unwell and possibly would have forgotten until prompted.
